lexerOfRealNumbers/lexerOfRealNumbers/lexer.c

```diff
@@ -0,0 +1,57 @@
+#include "lexer.h"
+#include <ctype.h>
+#include <string.h>
+
+bool isTheSequenceANumber(const char* string) {
+	size_t sizeString = strlen(string);
+	int state = 0;
+	for (size_t i = 0; i < sizeString; ++i) {
+		switch (state)
+		{
+		case(0):
+			if (isdigit(string[i])) {
+				state = 1;
+				break;
+			}
+			return false;
+		case(1):
+			if (isdigit(string[i])) {
+				state = 1;
+				break;
+			} else if (string[i] == '.') {
+				state = 2;
+				break;
+			} else if (string[i] == 'E') {
+				state = 3;
+				break;
+			}
+			return false;
+		case(2):
+			if (string[i] == 'E' && string[i-1] != '.') {
```

Comment:
```suggestion
			if (string[i] == 'E' && string[i - 1] != '.') {
```
Но вообще нет, автомат так не умеет. `string[i - 1]`, как и другие факты из прошлого, кодируются состояниями. Иначе таблица переходов была бы трёхмерной.